Sources/RxMoya/RxMoyaProvider.swift

```diff
@@ -6,22 +6,35 @@ import Moya
 
 /// Subclass of MoyaProvider that returns Observable instances when requests are made. Much better than using completion closures.
 open class RxMoyaProvider<Target>: MoyaProvider<Target> where Target: TargetType {
+    
+    /// Propagated to Alamofire as callback queue. If nil - main queue will be used.
+    fileprivate let queue: DispatchQueue?
+    
     /// Initializes a reactive provider.
-    override public init(endpointClosure: @escaping EndpointClosure = MoyaProvider.defaultEndpointMapping,
+    public init(endpointClosure: @escaping EndpointClosure = MoyaProvider.defaultEndpointMapping,
```

Comment:
Yes, I think adding it to all 3 is the best approach, so we have feature parity (as much as possible) between all three.
We also might be able to remove all the `queue` logic from our Reactive and Rx subclasses by instead implementing it in MoyaProvider and then doing super.request instead of self.request in places like https://github.com/Moya/Moya/pull/1055/files#diff-ffd4342ff685cb2dff7f9c3cf77cef22R38

Let's do the deprecate approach, and then maybe in 9.0.0 I can remove the deprecated function so we keep our codebase clean (I can make that PR after we're done with this one)